packages/blocks/src/api/serializer.js

```diff
@@ -137,6 +137,9 @@ export function getSaveElement(
 
 	blockPropsProvider.blockType = blockType;
 	blockPropsProvider.attributes = attributes;
+	// Debugging: Empty InnerBlocks on Site Editor, while it has items on post Editor.
+	// Having the same block structure on both site and post editor.
+	// Delete these lines before merging the PR when discovered the reason.
```

Comment:
Added this line in case anyone wants to debug and discover why is it updating the code with the rename on the post editor but not on the site editor. As is mentioned [here] (https://github.com/WordPress/gutenberg/pull/40506#issuecomment-1122391822)

I will keep taking a look at it tomorrow ( 21st June, for the async 😄  )

Reply:
Thanks Carlos! FWIW, I think it'd be fine to consider this PR for merging even if the Site Editor isn't updating the block IDs. @DAreRodz and I were running into the same issue today while working on https://github.com/WordPress/gutenberg/pull/41807, so I think it's pretty much pre-existing and unrelated to what we're doing here. We might consider fixing it, but maybe in a separate PR so we don't block this one 🙂

Reply:
Comment reverted!